src/receive_imf.rs

```diff
@@ -1629,7 +1631,7 @@ RETURNING id
                     } else {
                         DownloadState::Done
                     },
-                    mime_parser.hop_info
+                    if trash || hidden { "" } else { &mime_parser.hop_info },
```

Comment:
Maybe you can pull this improvement out into a separate PR that also updates `MsgId::trash()` and `delete_expired_messages()`?